### PR TITLE
Toyota: match stock standstill entrance behavior

### LINKED ISSUE #1158: Toyota: match stock standstill entrance behavior
for cars not in `NO_STOP_TIMER_CAR`. One attempt: https://github.com/commaai/openpilot/pull/26721

Might want to clean up the `if CS.pcm_acc_status != 8:` stuff, why not just check CS.standstill?

### LINKED ISSUE #29613: Toyota: match stock standstill entrance behavior
for cars not in `NO_STOP_TIMER_CAR`. One attempt: https://github.com/commaai/openpilot/pull/26721

Might want to clean up the `if CS.pcm_acc_status != 8:` stuff, why not just check CS.standstill?

### PR DESCRIPTION
**Pull request by cydia2020**
**Description**

*Behavior explained:*
This PR fixes commaai/opendbc#1158 by mimicking stock longitudinal standstill entrance behavior. After this PR is merged, openpilot will:
- Enter standstill once the car determines that it is appropriate (vehicle is stopped, standstill bit is always set unless openpilot wants to resume).
- Request exiting standstill once the lead starts moving (CC.cruiseControl.resume = True).
- Keep requesting to exit standstill for 1.5 seconds after the last CC.cruiseControl.resume = True request.
  - This timer resets after the vehicle exits standstill to prevent PCM from not re-entering standstill.
- Request resume for as long as the lead is creeping if vehicle is stopped and lead is creeping, it will stop doing so once the lead stops moving and long planner isn't requesting any acceleration.
  - The user can elect to not depress the RES button, keeping the vehicle in standstill, and only resume once the user deems it appropriate.
- Request resume exit immediately if the user engages at a standstill with a far-away lead or with no lead.
- Ignore standstill on `NO_STOP_TIMER_CAR`

**Verification**

Went on a drive and vehicle behaves like stock.

**Route**

7069d98b0de453c0|2024-02-08--12-21-10--0
7069d98b0de453c0|2024-02-08--11-36-39--0

See bookmarks, there are a few good examples in there about how this is now handled, and this should be pretty easy to test as well.

### CODE REVIEW

**Comment by cydia2020** [comment]
One issue with this is the event that's created:
`EventName.resumeRequired` will always be displayed even if the vehicle isn't ready for resume, should this be fixed? It can definitely be by using
```python
if ret.cruiseState.standstill and not ret.brakePressed and not self.CP.enableGasInterceptor and \
         c.cruiseControl.resume
```
But then the same hysteresis that's used for the vehicle won't apply to the alert.

File: selfdrive/car/toyota/carcontroller.py [hunk 1]
```diff
@@ -121,14 +125,19 @@ def update(self, CC, CS, now_nanos):
     if not CC.enabled and CS.pcm_acc_status:
       pcm_cancel_cmd = 1
 
-    # on entering standstill, send standstill request
-    if CS.out.standstill and not self.last_standstill and (self.CP.carFingerprint not in NO_STOP_TIMER_CAR or self.CP.enableGasInterceptor):
-      self.standstill_req = True
-    if CS.pcm_acc_status != 8:
-      # pcm entered standstill or it's disabled
-      self.standstill_req = False
```
Comment (sshane) [inline selfdrive/car/toyota/carcontroller.py:129]: Why remove this?

Comment (cydia2020) [reply selfdrive/car/toyota/carcontroller.py:129]: It seems unnecessary, we can request standstill even if cruise is off and it wouldn't cause any issues.

**Comment by cydia2020** [comment]
Hey folks, would this something that's mergeable?

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**